PhysicsTools/PatAlgos/plugins/PATMuonSlimmer.cc

```diff
@@ -44,12 +44,22 @@ namespace pat {
 
 pat::PATMuonSlimmer::PATMuonSlimmer(const edm::ParameterSet & iConfig) :
     src_(consumes<pat::MuonCollection>(iConfig.getParameter<edm::InputTag>("src"))),
-    pf_(mayConsume<reco::PFCandidateCollection>(iConfig.getParameter<edm::InputTag>("pfCandidates"))),
-    pf2pc_(mayConsume<edm::Association<pat::PackedCandidateCollection>>(iConfig.getParameter<edm::InputTag>("packedPFCandidates"))),
     linkToPackedPF_(iConfig.getParameter<bool>("linkToPackedPFCandidates")),
     saveTeVMuons_(iConfig.getParameter<std::string>("saveTeVMuons")),
     modifyMuon_(iConfig.getParameter<bool>("modifyMuons"))
 {
+    if (linkToPackedPF_) {
+        const std::vector<edm::InputTag> & pf = (iConfig.existsAs<std::vector<edm::InputTag>>("pfCandidates") ?
+                    iConfig.getParameter<std::vector<edm::InputTag>>("pfCandidates") :
+                    std::vector<edm::InputTag>(1,iConfig.getParameter<edm::InputTag>("pfCandidates")));
```

Comment:
looks like the configs should just change to pfCandidates be a vector and this type detection be removed